vividus-plugin-web-app/src/main/java/org/vividus/proxy/steps/ProxySteps.java

```diff
@@ -96,11 +95,16 @@ public void clearProxyLog()
      */
     @Then("number of HTTP $httpMethods requests with URL pattern `$urlPattern` is $comparisonRule `$number`")
     public List<HarEntry> checkNumberOfRequests(Set<HttpMethod> httpMethods, Pattern urlPattern,
-            ComparisonRule comparisonRule, long number) throws IOException
+            ComparisonRule comparisonRule, int number) throws IOException
     {
         List<HarEntry> harEntries = getLogEntries(httpMethods, urlPattern);
-        assertSize(String.format("Number of HTTP %s requests matching URL pattern '%s'",
-                methodsToString(httpMethods, ", "), urlPattern), harEntries, comparisonRule, number);
+        String description = String.format("Number of HTTP %s requests matching URL pattern '%s'",
+                methodsToString(httpMethods, ", "), urlPattern);
+        if (!softAssert.assertThat(description, harEntries.size(), comparisonRule.getComparisonRule(number)))
+        {
+            byte[] harBytes = OBJECT_MAPPER.writeValueAsBytes(proxy.getRecordedData());
+            attachmentPublisher.publishAttachment(harBytes, "har.har");
```

Comment:
maybe something like Network Logs.har ?

Reply:
makes sense, but it should go as a separate commit

PS: the goal of this PR is to make new logic of proxy check clearer